test/dnsimple/domain_delegation_signer_records_spec.js

```diff
@@ -0,0 +1,209 @@
+
+      endpoint.done();
+      done();
+    });
+
+    it('supports extra request options', function(done) {
+      var endpoint = nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records?foo=bar')
+        .reply(fixture.statusCode, fixture.body);
+
+      dnsimple.domains.listDelegationSignerRecords(accountId, domainId, {query: {foo: 'bar'}});
+
+      endpoint.done();
+      done();
+    });
+
+    it('supports sorting', function(done) {
+      var endpoint = nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records?sort=from%3Aasc')
+        .reply(fixture.statusCode, fixture.body);
+
+      dnsimple.domains.listDelegationSignerRecords(accountId, domainId, {sort: 'from:asc'});
+
+      endpoint.done();
+      done();
+    });
+
+    it('produces an delegation signer records list', function(done) {
+      nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records')
+        .reply(fixture.statusCode, fixture.body);
+
+      dnsimple.domains.listDelegationSignerRecords(accountId, domainId).then(function(response) {
+        var dsRecords = response.data;
+        expect(dsRecords.length).to.eq(1);
+        done();
+      }, function(error) {
+        done(error);
+      });
+    });
+
+    it('exposes the pagination info', function(done) {
+      nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records')
+        .reply(fixture.statusCode, fixture.body);
+
+      dnsimple.domains.listDelegationSignerRecords(accountId, domainId).then(function(response) {
+        var pagination = response.pagination;
+        expect(pagination).to.not.be.null;
+        expect(pagination.current_page).to.eq(1);
+        done();
+      }, function(error) {
+        done(error);
+      });
+    });
+  });
+
+  describe('#allDelegationSignerRecords', function() {
+    var accountId = '1010';
+    var domainId = 'example.com'
+
+    it('produces a complete list', function(done) {
+      var fixture1 = testUtils.fixture('pages-1of3.http');
+      nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records?page=1')
+        .reply(fixture1.statusCode, fixture1.body);
+
+      var fixture2 = testUtils.fixture('pages-2of3.http');
+      nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records?page=2')
+        .reply(fixture2.statusCode, fixture2.body);
+
+      var fixture3 = testUtils.fixture('pages-3of3.http');
+      nock('https://api.dnsimple.com')
+        .get('/v2/1010/domains/example.com/ds_records?page=3')
+        .reply(fixture3.statusCode, fixture3.body);
+
+      dnsimple.domains.allDelegationSignerRecords(accountId, domainId).then(function(items) {
+        expect(items.length).to.eq(5);
+        expect(items[0].id).to.eq(1);
+        expect(items[4].id).to.eq(5);
+        done();
+      }, function(error) {
+        done(error);
+      }).catch(function(error) {
```

Comment:
I think it's not quite the same, one being a rejected promise, the other being an actual exception.